README.md

```diff
@@ -82,66 +76,87 @@ The `samlconf` script may take the following parameters:
+[KeycloakIdpSSOResponderProvider](external/implementations/samlconf-keycloak-impl/src/main/kotlin/org/codice/compliance/saml/plugin/keycloak/KeycloakIdpSSOResponderProvider.kt) can be used as examples.
+1. Package that file into a jar.
+1. Place the jar from step 1 and the IdP's metadata into a directory.
+1. Setup your IdP.
+1. Configure your IdP with the test kit's [SP metadata](deployment/distribution/src/main/resources/samlconf-sp-metadata.xml).
+1. The directory from step 3 should be referred to with the `-i` option when running tests. (See [Run the tests](#run-the-tests))
 
-##### samlconf-ddf-impl
-Plugin and IdP metadata XML for the ddf implementation of IdP. See the [DDF](external/implementations/samlconf-ddf-impl/README.md) in this directory for details.
+#### Formatting
+If during development the build fails due to `format violations`, run `gradlew spotlessApply` to fix the formatting.
 
-##### samlconf-keycloak-impl
-Plugin and IdP metadata XML for the Keycloak implementation of IdP. See the [Keycloak](external/implementations/samlconf-keycloak-impl/README.md) in this directory for details.
-
-### deployment
-This module is the project's full package deployment.
-
-#### distribution
-This module contains all the runtime elements including scripts, jars, and configurations.
+#### Project Structure
+This section will briefly talk about the project structure.
 
-#### docker
-This module contains the logic for building a docker image.
-To build this module you must run the docker task by executing `gradlew build docker`.
+```
+.
+├── ctk - test related modules
+│   │
+│   ├── common - classes relating to utility and verification of the test classes
+│   │               NOTE: Schema verification is run before tests and verifications.
+│   │
+│   └── idp - tests being written against a SAML IdP. The `src` directory of the module is organized by the SAML specification as follows:
+│              * Package: Based on Profile (i.e. WebSSO, Single Logout)
+│                * Class: Based on Binding (i.e. POST, REDIRECT, ARTIFACT)
+│
+├── deployment - the project's full package deployment
+│   │
+│   ├── distribution - runtime elements including scripts, jars, and configurations
+│   │
+│   └── docker - logic for building a Docker image
+│
+├── external - files related to a specific SAML implementing product
+│   │
+│   ├── samlconf-plugins-api -  API that must be implemented for a SAML product in order to run this test kit against that product
+│   │
+│   └── implementations - implementations of the API for specific SAML products
+│       │
+│       ├── samlconf-ddf-impl - plugin and IdP metadata XML for the ddf implementation of IdP
+│       │
+│       └── samlconf-keycloak-impl - plugin and IdP metadata XML for the Keycloak implementation of IdP
+│
+└── library - Java classes that have been copied over from DDF to support operations that shouldn't be handled by the test code;
```

Comment:
❗️ Change to
`Java classes copied from DDF to support operations outside the scope of the test code, e.g. signature validation using x509 certificates.`